admin/broadleaf-open-admin-platform/src/main/java/org/broadleafcommerce/openadmin/web/controller/entity/AdminBasicEntityController.java

```diff
@@ -416,10 +416,38 @@ public String saveEntityJson(HttpServletRequest request, HttpServletResponse res
         if (result.hasErrors()) {
             populateJsonValidationErrors(entityForm, result, json);
         }
+        List<String> dirtyList = buildDirtyList(pathVars, request, id);
+        if (!dirtyList.isEmpty()) {
```

Comment:
Based on `buildDirtyList()` below, if you catch a `ServiceException` you just return null. If that happens, this line of code will throw a `NullPointerException` because `dirtyList` is null.

You should get into the habit of using `CollectionUtils` as much as possible. In this case, this code should transform to:

```java
if (CollectionUtils.isNotEmpty(dirtyList))
```

Which gives you a couple of advantages:

1. Readability is 100x better
2. It automatically protects against NPEs. If `dirtyList` is null, then that method will return `false` (it treats `null` as empty)